### PR TITLE
feat: Created backend repo, added handlers, logic.

### PR DESCRIPTION
**Pull request by lexBogos**
poiproduct-service serverless.yml contains configuration for 2 lambda functions, API is not working at all, but YAML configuration is correct (+)
The getProductsList OR getProductsById lambda function returns a correct response (POINT1)(+)
The getProductsById AND getProductsList lambda functions return a correct response code (POINT2)(+)
Your own Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. AND POINT1 and POINT2 are done.(-)
 ES6 modules are used for product-service implementation(+)
 Webpack is configured for product-service(+)
 SWAGGER documentation is created for product-service(+)
Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered) (You may use JEST)(-)
 Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.(+)
 Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).(+)

https://z9f5582xt9.execute-api.eu-west-1.amazonaws.com/dev/products/
https://z9f5582xt9.execute-api.eu-west-1.amazonaws.com/dev/products/{productId}

total: 9/11

### CODE REVIEW

**Review by maksumov — COMMENTED** [review]
Просто для сведения (не планирую снижать за это оценку):
- https://z9f5582xt9.execute-api.eu-west-1.amazonaws.com/dev/products/7567ec4b-b10c-48c5-9345-fc73c48a80aa - отрабатывает
- https://z9f5582xt9.execute-api.eu-west-1.amazonaws.com/dev/products/11 - валится с ошибкой 500

**Comment by maksumov** [comment]
### Main tasks check

- [x] `serverless.yaml` for product-service contains configuration for 2 lambda functions, YAML configuration is correct
- [x] The getProductsList OR getProductsById lambda function returns a correct response ([POINT1](https://wnxy68e0x6.execute-api.eu-west-1.amazonaws.com/dev/product/available/))
- [x] The getProductsById AND getProductsList lambda functions return a correct response code ([POINT2](https://wnxy68e0x6.execute-api.eu-west-1.amazonaws.com/dev/product/available/559725))
- [ ] Your own Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. AND POINT1 and POINT2 are done.
 
### Additional (optional) tasks check

- [x] Async/await is used in lambda functions
- [x] ES6 modules are used for product-service implementation
- [x] Webpack is configured for product-service
- [x] SWAGGER documentation is created for product-service
- [ ] Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered)
- [x] Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
- [x] Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers)

Tasks completed: 9/11

**Review by maksumov — APPROVED** [review]

File: product-service/src/adapters/generate-response.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    ? {
+        statusCode: 200,
+        body: JSON.stringify(returnedObj),
+      }
+    : {
+        statusCode: 200,
+        body: `The product with id:'${resultProduct.id}' not found`,
+      };
+
+  return result;
```
[the remark this answers is not in the document]
Comment (numatay) [reply product-service/src/adapters/generate-response.js:13]: Я вот тоже не понимаю, мы же запрашиваем определенный ресурс по `id`. Я считаю что `http` код должен быть `404`

File: product-service/src/handlers/get-product-by-id.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import productFilter from '../utils/product-filter';
+import generateResponse from '../adapters/generate-response';
+import mockDataProvider from '../data-providers/mock-data-provider';
+
+export const getProductById = async event => {
+  console.log('Lambda invocation with event: ', event);
+  try {
+    const eventProductId = event && event.pathParameters && event.pathParameters.productId;
```
Comment (numatay) [inline product-service/src/handlers/get-product-by-id.js:8]: В typeScript-e можно вот так делать `event?.pathParameters?.productId`

File: product-service/src/adapters/generate-response.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (lexBogos) [reply product-service/src/adapters/generate-response.js:13]: Да верно, нужно 404 отдавать

File: product-service/src/handlers/get-product-by-id.js (hunk 2 above)
Comment (lexBogos) [reply product-service/src/handlers/get-product-by-id.js:8]: Спасибо, буду знать)

**Pull request merged**